pkg/cvo/internal/generic.go

```diff
@@ -29,6 +30,32 @@ func readUnstructuredV1OrDie(objBytes []byte) *unstructured.Unstructured {
 	return udi.(*unstructured.Unstructured)
 }
 
+func deleteUnstructured(ctx context.Context, client dynamic.ResourceInterface, required *unstructured.Unstructured) (bool, error) {
+	if required.GetName() == "" {
+		return false, fmt.Errorf("Error running delete, invalid object: name cannot be empty")
+	}
+	if delAnnoFound, err := resourcedelete.ValidDeleteAnnotation(required.GetAnnotations()); !delAnnoFound || err != nil {
+		return delAnnoFound, err
+	}
+	resource := resourcedelete.Resource{
+		Kind:      "unstructured",
```

Comment:
Shouldn't we pull this out of `required`?

Reply:
Yes. Otherwise I could get collisions at the Namespace level I believe. Corrected.